packages/server/graphql/private/mutations/processScheduledLocks.ts

```diff
@@ -0,0 +1,107 @@
+import ms from 'ms'
+import sendTeamsLimitEmail from '../../../billing/helpers/sendTeamsLimitEmail'
+import getRethink from '../../../database/rethinkDriver'
+import isValid from '../../isValid'
+import {MutationResolvers} from '../resolverTypes'
+
+const processScheduledLocks: MutationResolvers['processScheduledLocks'] = async (
+  _source,
+  _args,
+  {dataLoader}
+) => {
+  const r = await getRethink()
+  const now = new Date().getTime()
+  const inOneWeek = now + ms('7d')
+  const inSixDays = now + ms('6d')
+
+  const orgsScheduledForLockWithinWeek = await r
```

Comment:
As the mutation is run daily, there shouldn't be many `orgsScheduledForLockWithinWeek`. It also doesn't matter too much if the mutation takes a while to execute because it's a chron job. So, while there's a lot of mapping and filtering below, I _aimed_ for readability over performance, although it may still be difficult to read 🤷‍♂️